render/renderer.go

```diff
@@ -58,11 +59,16 @@ type Renderer struct {
 	Result    *image.NRGBA // The image result after rendering using the Render functions.
 	tileCache map[uint32]image.Image
 	engine    RendererEngine
+	fs        fs.FS
 }
 
 // NewRenderer creates new rendering engine instance.
 func NewRenderer(m *tiled.Map) (*Renderer, error) {
-	r := &Renderer{m: m, tileCache: make(map[uint32]image.Image)}
+	return NewRendererWithFileSystem(m, nil)
+}
+
+func NewRendererWithFileSystem(m *tiled.Map, fs fs.FS) (*Renderer, error) {
```

Comment:
Please add documentation comment for this function

Reply:
ff45cc6dc99527a4df6037c84418ef8a6086dcb8 :)